### DISCUSSION TITLE
Set a DTSTAMP for new VEVENT/VTODO components

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [collective/icalendar#542] [Ideas]
VTODO (amongst others) components MUST a DTSTAMP entry, and it should contain the timestamp of the components creation.

I'd like to propose that creating new components of these types have this value automatically added, with `datetime.now` as a default.

This will basically make these components standards-compliant (and the value is an expected one too).

**Comment by geier** [comment]
icalendar's current approach is to let the user handle _everything_ themselves. Convenience functions, like the one suggested here, could go hand in hand with implementing a validation checker (as mentioned in #28). This could be a point for the 4.0 roadmap (#186) if anyone is interested in working on it.

**Comment by untitaker** [comment]
My position is that icalendar really shouldn't let the user produce invalid data by default. A validation checker is opt-in.
